js/client/modules/@arangodb/testsuites/gtest.js

```diff
@@ -136,37 +150,43 @@ function gtestRunner (testname, options) {
       // all non gtest args have to come last
       argv.push('--log.line-number');
       argv.push(options.extremeVerbosity ? "true" : "false");
-      results.basics = pu.executeAndWait(run, argv, options, 'all-gtest', rootDir, options.coreCheck);
-      results.basics.failed = results.basics.status ? 0 : 1;
-      if (!results.basics.status) {
+      results[name] = pu.executeAndWait(binary, argv, options, 'all-gtest', rootDir, options.coreCheck);
+      results[name].failed = results[name].status ? 0 : 1;
+      if (!results[name].status) {
         results.failed += 1;
       }
       results = getGTestResults(testResultJsonFile, results);
       tmpMgr.destructor((results.failed === 0) && options.cleanup);
     } else {
       results.failed += 1;
-      results.basics = {
+      results[name] = {
         failed: 1,
         status: false,
-        message: 'binary "arangodbtests" not found when trying to run suite "all-gtest"'
+        message: `binary ${testfilename} not found when trying to run suite "all-gtest"`
       };
     }
   }
   return results;
 }
 
 exports.setup = function (testFns, defaultFns, opts, fnDocs, optionsDoc, allTestPaths) {
-  Object.assign(allTestPaths, testPaths);
-  testFns['gtest'] = x => gtestRunner('arangodbtests', x);
-  testFns['catch'] = x => gtestRunner('arangodbtests', x);
-  testFns['boost'] = x => gtestRunner('arangodbtests', x);
-
   opts['skipGtest'] = false;
-  opts['skipGeo'] = false; // not used anymore - only here for downwards-compatibility
 
-  defaultFns.push('gtest');
+  Object.assign(allTestPaths, testPaths);
 
-  testFns['gtest_replication2'] = x => gtestRunner('arangodbtests_replication2', x);
+  const tests = [ 'arangodbtests_zkd' ];
+
+  for(const test of tests) {
+    testFns[test] = x => gtestRunner(test, test, x);
+  }
+  testFns['gtest'] = x => gtestRunner('arangodbtests', 'arangodbtests', x);
+
+  testFns['gtest_replication2'] = x => gtestRunner('arangodbtests_replication2', 'gtest-replication2', x);
+  
+  let iresearch_filter = ['gtest_filter=IResearch*'];
+  testFns['gtest_iresearch'] = x => gtestRunner('arangodbtests', 'gtest-iresearch', x, iresearch_filter);
+  let no_iresearch_filter = ['gtest_filter=-IResearch*:LongRunning*'];
+  testFns['gtest_arangodb'] = x => gtestRunner('arangodbtests', 'gtest-arangodb', x, no_iresearch_filter);
```

Comment:
I think that all test cases can be located in `const tests` array?

Reply:
tests iterates over binaries to run; we want to have different options. but yes, could probably be done - but rather later for devel and not for the backport.